src/sentry/api/serializers/models/organization.py

```diff
@@ -486,29 +486,23 @@ def get_attrs(
         return super().get_attrs(item_list, user)
 
     def _project_list(self, organization: Organization, access: Access) -> list[Project]:
-        member_projects = list(access.projects)
-        member_project_ids = [p.id for p in member_projects]
-        other_projects = list(
-            Project.objects.filter(organization=organization, status=ProjectStatus.VISIBLE).exclude(
-                id__in=member_project_ids
-            )
+        project_list = list(
```

Comment:
Mark made an excellent point in my PR about this logic:  we actually do need to split between those the user is a member of and those they are not, due to the need to order first by membership then by slug.

That should be possible using the access.project_ids_with_team_membership and friends.

Reply:
It doesn't look like the existing code is grouping by membership. Doesn't

```python
project_list = sorted(other_projects + member_projects, key=lambda x: x.slug)
```

combine the groups together and sort everything by slug?

Reply:
You're totally right!  This logic was written weird; why on earth wouldn't it just make the query the way you did?  Huh.  Ok!  Nice catch.

Reply:
Don't know, but I speculated about it at https://github.com/getsentry/sentry/pull/41002/files/8b4f503784b29787852094617987d5f67d0107a4#r1014508570